### DISCUSSION TITLE
Cannot register or upload any MacOS ISO

### DISCUSSION THREAD

**Discussion by MejdiB** [apache/cloudstack#9786] [Q&A]
The register or upload of MacOS ISO files seems not to be supported by Cloudstack. With every MacOS version from v11-v15 the same error occurs: 

`Failed to install: Template content is unsupported, or mismatch between selected format and template content. Found : Apple Driver Map, blocksize 512, blockcount 15728640, devtype 0, devid 0, driver count 0, contains[@0x200]`

Is this a bug?

**Comment by rajujith** [comment]
@MejdiB  Is the ISO in 'ISO 9660' format? if not is it possible to convert it to this format and try?

**Reply by MejdiB** [reply]
@rajujith Thank you very much. It was indeed the issue and now I am able to download the ISO files from my local server. However, when starting a VM with a mac os 11 iso, I get the error message "No bootable device", I assume I have to install some KVM plugins?

**Reply by rajujith** [reply]
@MejdiB "No bootable device" could be related to UEFI? You may want to configure that if it has not been done already.

**Reply by MejdiB** [reply]
@rajujith I have tried with MacOS 11 and 15 with UEFI and SECURE, here is the libvirt exception: `org.libvirt.LibvirtException: unsupported configuration: IDE controllers are unsupported for this QEMU binary or machine type`. It has to be said that `macOS 11` is mapped as the OS to the ISO files. When I change this to another OS mapping with SATA controller (i.e. Windows, I get the same error as above)
